airbyte-integrations/connectors/source-zendesk-support/source_zendesk_support/streams.py

```diff
@@ -0,0 +1,422 @@
+        return None
+
+    @staticmethod
+    def str2datetime(s):
+        """convert string to datetime object"""
+        if not s:
+            return None
+        return datetime.strptime(s, DATETIME_FORMAT)
+
+    @staticmethod
+    def datetime2str(dt):
+        """convert datetime object to string"""
+        return datetime.strftime(dt.replace(tzinfo=pytz.UTC), DATETIME_FORMAT)
+
+
+class UserSettingsStream(SourceZendeskSupportStream):
+    """Stream for checking of a request token and permissions"""
+
+    def path(self, *args, **kwargs) -> str:
+        return "account/settings.json"
+
+    def next_page_token(self, *args, **kwargs) -> Optional[Mapping[str, Any]]:
+        # this data without listing
+        return None
+
+    def parse_response(self, response: requests.Response, **kwargs) -> Iterable[Mapping]:
+        """returns data from API"""
+        yield from [response.json().get("settings") or {}]
+
+    def get_settings(self) -> Tuple[Mapping[str, Any], Union[str, None]]:
+        for resp in self.read_records(SyncMode.full_refresh):
+            return resp, None
+        return None, "not found settings"
+
+
+class IncrementalBasicSearchStream(SourceZendeskSupportStream, ABC):
+    """Base class for all data lists with a incremental stream"""
+
+    # max size of one data chunk. 100 is limitation of ZenDesk
+    state_checkpoint_interval = SourceZendeskSupportStream.page_size
+
+    # default sorted field
+    cursor_field = SourceZendeskSupportStream.updated_at_field
+
+    def __init__(self, start_date: str, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        # add the custom value for skiping of not relevant records
+        self._start_date = self.str2datetime(start_date) if isinstance(start_date, str) else start_date
+
+    def next_page_token(self, response: requests.Response) -> Optional[Mapping[str, Any]]:
+        next_page = self._parse_next_page_number(response)
+        if next_page:
+            return {"next_page": next_page}
+        return None
+
+    def parse_response(self, response: requests.Response, **kwargs) -> Iterable[Mapping]:
+        # Root of all responses of searching endpoints is 'results'
+        yield from response.json()["results"] or []
+
+    def path(self, *args, **kargs) -> str:
+        return "search.json"
+
+    def request_params(
+        self, stream_state: Mapping[str, Any] = None, next_page_token: Mapping[str, Any] = None, **kwargs
+    ) -> MutableMapping[str, Any]:
+        updated_after = None
+        if stream_state and stream_state.get(self.cursor_field):
+            updated_after = self.str2datetime(stream_state[self.cursor_field])
+
+        # add the 'query' parameter
+        conds = [f"type:{self.entity_type[:-1]}"]
+        conds.append("created>%s" % self.datetime2str(self._start_date))
+        if updated_after:
+            conds.append("updated>%s" % self.datetime2str(updated_after))
+
+        res = {
+            "query": " ".join(conds),
```

Comment:
could you add comment with  example of real  query

Reply:
this logic was removed